### ISSUE TITLE
Sam delete does not honor parameters in config-env.

### ISSUE THREAD

**Issue by matthewhembree** [aws/aws-sam-cli#3514]
### Description:

When running `sam delete`, the region specified in config-env is not honored.

### Steps to reproduce:

samconfig.toml:
```toml
[dev]
[dev.global]
[dev.global.parameters]
region = "us-west-2"
[dev.delete]
[dev.delete.parameters]
region = "us-west-2"
[dev.deploy]
[dev.deploy.parameters]
region = "us-west-2"
capabilities = "CAPABILITY_IAM"
parameter_overrides = "StackTagName=\"my-stack\" EnvironmentTagName=\"dev\" EnvironmentAPIGatewayStageName=\"default\""
image_repositories = []
stack_name = "my-stack-dev"
```

```
sam delete --stack-name my-stack-dev --config-env dev --no-prompts --debug
```

### Observed result:

```
sam delete --stack-name my-stack-dev --config-env dev --no-prompts --debug
2021-12-03 21:29:16,924 | Using config file: samconfig.toml, config environment: dev
2021-12-03 21:29:16,924 | Expand command line arguments to:
2021-12-03 21:29:16,924 | --stack_name=my-stack --no_prompts
2021-12-03 21:29:17,022 | Config file location: /Users/username/Documents/gitFolder/my-stack/samconfig.toml
2021-12-03 21:29:17,022 | Loading configuration values from [dev.['deploy'].parameters] (env.command_name.section) in config file at '/Users/username/Documents/gitFolder/my-stack/samconfig.toml'...
2021-12-03 21:29:17,043 | Configuration values successfully loaded.
2021-12-03 21:29:17,043 | Configuration values are: {'region': 'us-west-2', 'capabilities': 'CAPABILITY_IAM', 'parameter_overrides': 'StackTagName="my-stack" EnvironmentTagName="dev" EnvironmentAPIGatewayStageName="default"', 'image_repositories': [], 'stack_name': 'my-stack-dev'}
Usage: sam delete [OPTIONS]

Error: Missing option '--region', region is required to run the non guided delete command.
```

### Expected result:

The stack to be deleted.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Mac, Ubuntu
2. `sam --version`: Mac: SAM CLI, version 1.35.0 && Ubuntu: SAM CLI, version 1.35.0
3. AWS region: us-west-2

`Add --debug flag to command you are running`

**Comment by matthewhembree** [comment]
Closing. `sam delete` does *honor the config-env. My output was sanitized and did not reflect a `--stack-name` that differed from what was configured in the config-env. Therefore, [this block](https://github.com/aws/aws-sam-cli/blob/2b2a921172a22f43ef8580eb8ad0be1558e9cb7d/samcli/commands/delete/delete_context.py#L92) didn't run. Probably as a safety measure.

*`sam delete` does have unorthodox behavior where it uses `env.deploy.parameters` rather than the anticipated `env.delete.parameters`, but when you think about it it sort of makes sense. For me, I am using a modified version of the `sam pipeline` output. My feature branch deploy uses the same config-env as my stage1 config-env. The feature branch is deployed with `--stack-name`, so that will cause `sam delete --stack-name my-feature-stack --config-env stage1` to fail.

My debugging also leads me to believe that `env.global.parameters` will not parsed. As the [CONFIG_COMMAND](https://github.com/aws/aws-sam-cli/blob/2b2a921172a22f43ef8580eb8ad0be1558e9cb7d/samcli/commands/delete/delete_context.py#L81) references [only](https://github.com/aws/aws-sam-cli/blob/2b2a921172a22f43ef8580eb8ad0be1558e9cb7d/samcli/commands/delete/delete_context.py#L30) `env.deploy`.


**TL;DR:** `--stack-name` did not match `env.deploy.parameters.stack_name`
